src/main/java/com/example/demo/Controllers/MovieController.java

```diff
@@ -1,24 +1,70 @@
 package com.example.demo.Controllers;
 
 import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.validation.Valid;
+
+import java.net.URI;
+import java.net.URISyntaxException;
 
 import com.example.demo.Models.Movie;
-import com.example.demo.Repositories.MovieRepository;
+import com.example.demo.Services.MovieService;
+
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
 
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.*;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.http.ResponseEntity;
+import org.springframework.hateoas.IanaLinkRelations;
+
 @RestController
 public class MovieController {
-    private final MovieRepository repository;
 
-    MovieController(MovieRepository repository) {
-        this.repository = repository;
-    }
+    @Autowired
+    private MovieService movieService;
 
     // Aggregate root
 
+    @GetMapping("/api/movies/{id}")
+    EntityModel<Movie> one(@PathVariable Long id) {
+
+        Movie movie = movieService.findById(id);
+
+        return new EntityModel<>(movie, linkTo(methodOn(MovieController.class).one(id)).withSelfRel(),
+                linkTo(methodOn(MovieController.class).all()).withRel("movies"));
+    }
+
     @GetMapping("/api/movies")
-    List<Movie> all() {
-        return repository.findAll();
+    ResponseEntity<CollectionModel<EntityModel<Movie>>> all() {
+        List<EntityModel<Movie>> movies = movieService.findAll().stream()
+                .map(movie -> new EntityModel<>(movie,
+                        linkTo(methodOn(MovieController.class).one(movie.getId())).withSelfRel(),
+                        linkTo(methodOn(MovieController.class).all()).withRel("movies")))
+                .collect(Collectors.toList());
+
+        return ResponseEntity
```

Comment:
In the previous method you are using an _EntityModel_ as a return type, in this case you are using a _ResponseEntity_, Which are the differences between them?